cypress/e2e/models/migration/applicationinventory/assessment.ts

```diff
@@ -137,45 +138,49 @@ export class Assessment extends Application {
-                                $questionLine.text() ===
-                                    "How is the application clustering managed?"
-                            ) {
-                                optionToSelect = totalOptions - 1;
-                            } else {
-                                optionToSelect = Math.floor(totalOptions / 2);
-                            }
+        cy.get(assessmentBlock)
+            .its("length")
+            .then((count) => {
+                for (let i = 0; i < count - 1; i++) {
+                    cy.get(questionBlock).each(($question) => {
+                        let totalOptions = $question.find(stack).children("div").length;
+                        let optionToSelect: number;
+                        if (risk === "low") {
+                            optionToSelect = totalOptions - 1;
                             this.clickRadioOption($question, optionToSelect);
-                        });
-                } else {
-                    optionToSelect = 1;
-                    this.clickRadioOption($question, optionToSelect);
+                        } else if (risk === "medium") {
+                            cy.wrap($question)
+                                .children()
+                                .find("div.pf-v5-l-split__item")
+                                .then(($questionLine) => {
+                                    /* These 3 questions generate high risk with mean options,
+                                    hence to keep risk to medium, select last options for these set of specific questions */
+                                    if (
+                                        $questionLine.text() ===
+                                            "Does the application have legal and/or licensing requirements?" ||
+                                        $questionLine.text() ===
+                                            "Does the application require specific hardware?" ||
+                                        $questionLine.text() ===
+                                            "How is the application clustering managed?"
+                                    ) {
+                                        optionToSelect = totalOptions - 1;
+                                    } else {
+                                        optionToSelect = Math.floor(totalOptions / 2);
+                                    }
+                                    this.clickRadioOption($question, optionToSelect);
+                                });
+                        } else {
+                            optionToSelect = 1;
+                            this.clickRadioOption($question, optionToSelect);
+                        }
+                    });
+                    if (i === 4) {
+                        clickJs(commonView.nextButton);
+                    } else {
+                        clickJs(commonView.nextButton);
+                    }
```

Comment:
This method is too complex right now, some considerations:

1. `low` and `high` risk conditions are almost equal, so they can be combined with something like:


```typescript
if(risk === "medium") {
    ...
} else {
    this.clickRadioOption($question, risk === "high"  ?  1 : totalOptions - 1);
}
```

2. Those 3 special questions are making the if condition too long, and in the future some questions may change, be added or removed.

In order to make the code more maintainable, an idea could be to save the questions in an array at the start of the method and use `includes` instead of doing multipleConditions:

```typescript
const specialHighRiskQuestions = [
    "Does the application have legal and/or licensing requirements?",
    "Does the application require specific hardware?",
    "How is the application clustering managed?"
];
...
...

if(specialHighRiskQuestions.includes($questionLine.text())) {
    ...
}
```

3. The last if-else looks identical to me:

```typescript
 if (i === 4) {
    clickJs(commonView.nextButton);
} else {
    clickJs(commonView.nextButton);
}
```

Reply:
https://issues.redhat.com/browse/MTA-1736

Reply:
At least point 3 should be fixed before merging.

Reply:
all fo the points you mentioned are change worthy, but I dind't write any of them, I just wrapped it in a cy tag, and I don't intend to change anything in it for now, so I opened a ticket for it, and I'll work on it in the next PR

Reply:
@sshveta impletmented change, jenkins run passed: https://main-jenkins-csb-migrationqe.apps.ocp-c1.prod.psi.redhat.com/job/mta/job/mta-ui-tests-runner/1343/console